syne_tune/optimizer/schedulers/searchers/multi_fidelity_searcher.py

```diff
@@ -0,0 +1,151 @@
+import logging
+from collections import OrderedDict
+
+from typing import Dict, Any, Optional, List, Union
+
+from syne_tune.optimizer.schedulers.searchers.searcher import BaseSearcher
+from syne_tune.optimizer.schedulers.searchers.single_objective_searcher import (
+    SingleObjectiveBaseSearcher,
+)
+from syne_tune.optimizer.schedulers.searchers.searcher_factory import searcher_cls_dict
+
+logger = logging.getLogger(__name__)
+
+
+class IndependentMultiFidelitySearcher(BaseSearcher):
+    """
+    Searcher for the multi-fidelity setting which fits independent models for each
+    resource level as as proposed by Falkner et al.
+
+    | BOHB: Robust and Efficient Hyperparameter Optimization at Scale
+    | S. Falkner and A. Klein and F. Hutter
+    | Proceedings of the 35th International Conference on Machine Learning
+
+    # TODO: Update docstrings
+
+    :param config_space: Configuration space
+    :param points_to_evaluate: List of configurations to be evaluated
+        initially (in that order). Each config in the list can be partially
+        specified, or even be an empty dict. For each hyperparameter not
+        specified, the default value is determined using a midpoint heuristic.
+        If ``None`` (default), this is mapped to ``[dict()]``, a single default config
+        determined by the midpoint heuristic. If ``[]`` (empty list), no initial
+        configurations are specified.
+    """
+
+    def __init__(
+        self,
+        config_space: Dict[str, Any],
+        searcher_cls: Optional[Union[str, SingleObjectiveBaseSearcher]] = "kde",
+        points_to_evaluate: Optional[List[dict]] = None,
+        random_seed: Optional[int] = None,
+        searcher_kwargs: dict[str, Any] = None,
+    ):
+        super().__init__(
+            config_space,
+            points_to_evaluate=points_to_evaluate,
+            random_seed=random_seed,
+        )
+
+        if searcher_kwargs is None:
+            self.searcher_kwargs = dict()
+        else:
+            self.searcher_kwargs = searcher_kwargs
+
+        if isinstance(searcher_cls, str):
+            assert searcher_cls in searcher_cls_dict
+            self.searcher_cls = searcher_cls_dict.get(searcher_cls)
+        else:
+            self.searcher_cls = searcher_cls
+        self.models = OrderedDict()
```

Comment:
Is model the best name here? Perhaps `searchers` or `searchers_resource` are better names?
```suggestion
        self.searchers = OrderedDict()
```